In in-window mode, disable rewind on MSE videos and disable seeking and fast forward on live videos

danae404 opened the pull request:
#### dc7f1c2fadad4043a7ad33cf9e15092800805d29
<pre>
In in-window mode, disable rewind on MSE videos and disable seeking and fast forward on live videos
<a href="https://bugs.webkit.org/show_bug.cgi?id=274598">https://bugs.webkit.org/show_bug.cgi?id=274598</a>
<a href="https://rdar.apple.com/126924018">rdar://126924018</a>

Reviewed by Jer Noble.

This patch removes the rewind button in in-window mode if a video source type is not HLS or
a file. It also removes the fast forward button and disables seeking if a video&apos;s duration
is long enough to likely be a live video. This patch uses a threshold of a one week
(604800 seconds).

* Source/WebCore/Modules/mediacontrols/MediaControlsHost.cpp:
(WebCore::MediaControlsHost::inWindowFullscreen const):
(WebCore::MediaControlsHost::externalDeviceDisplayName const): Deleted.
(WebCore::MediaControlsHost::externalDeviceType const): Deleted.
(WebCore::MediaControlsHost::controlsDependOnPageScaleFactor const): Deleted.
(WebCore::MediaControlsHost::setControlsDependOnPageScaleFactor): Deleted.
(WebCore::MediaControlsHost::generateUUID): Deleted.
(WebCore::MediaControlsHost::shadowRootCSSText): Deleted.
(WebCore::MediaControlsHost::base64StringForIconNameAndType): Deleted.
(WebCore::MediaControlsHost::formattedStringForDuration): Deleted.
(): Deleted.
(WebCore::MediaControlsHost::showMediaControlsContextMenu): Deleted.
(WebCore::MediaControlsHost::sourceType const): Deleted.
* Source/WebCore/Modules/mediacontrols/MediaControlsHost.h:
* Source/WebCore/Modules/mediacontrols/MediaControlsHost.idl:
* Source/WebCore/Modules/modern-media-controls/media/media-controller.js:
(MediaController.prototype._updateControlsIfNeeded):

Canonical link: <a href="https://commits.webkit.org/279471@main">https://commits.webkit.org/279471@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/91cf0c414f35492064307a6a390e46c6450dc8cf

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/52996 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/32333 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/5483 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/56275 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/3719 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/55301 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/39230 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/3445 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/42988 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/2397 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/55094 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/30135 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/45749 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/24112 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/27151 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/3069 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/1878 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/49003 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/3226 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/57870 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/28137 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/3188 "Passed tests") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/50384 "Found 1 new test failure: imported/w3c/web-platform-tests/dom/events/scrolling/scrollend-event-not-fired-after-removing-scroller.tentative.html (failure)") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/29357 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/45966 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/49687 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/30276 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/7897 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/29111 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aestes commented on Source/WebCore/Modules/modern-media-controls/media/media-controller.js:
```diff
@@ -23,6 +23,8 @@
  * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
 
+const maxNonLiveDuration = 604800;
```
It would be more clear to express this like:
```
60 * 60 * 24 * 7
```
That makes it a little more obvious that the duration is 7 days (you could also add a comment).

aestes commented on Source/WebCore/Modules/mediacontrols/MediaControlsHost.cpp:
```diff
@@ -297,6 +297,12 @@ bool MediaControlsHost::inWindowFullscreen() const
+{
+    auto sourceType = this->sourceType();
+    if (sourceType)
+        return *sourceType == SourceType::HLS || *sourceType == SourceType::File;
```
Nit:
```
    if (auto sourceType = this->sourceType())
        return *sourceType == SourceType::HLS || *sourceType == SourceType::File;
```

danae404 committed:
In in-window mode, disable rewind on MSE videos and disable seeking and fast forward on live videos
https://bugs.webkit.org/show_bug.cgi?id=274598
rdar://126924018

Reviewed by Jer Noble.

This patch removes the rewind button in in-window mode if a video source type is not HLS or
a file. It also removes the fast forward button and disables seeking if a video's duration
is long enough to likely be a live video. This patch uses a threshold of a one week
(604800 seconds).

* Source/WebCore/Modules/mediacontrols/MediaControlsHost.cpp:
(WebCore::MediaControlsHost::inWindowFullscreen const):
(WebCore::MediaControlsHost::externalDeviceDisplayName const): Deleted.
(WebCore::MediaControlsHost::externalDeviceType const): Deleted.
(WebCore::MediaControlsHost::controlsDependOnPageScaleFactor const): Deleted.
(WebCore::MediaControlsHost::setControlsDependOnPageScaleFactor): Deleted.
(WebCore::MediaControlsHost::generateUUID): Deleted.
(WebCore::MediaControlsHost::shadowRootCSSText): Deleted.
(WebCore::MediaControlsHost::base64StringForIconNameAndType): Deleted.
(WebCore::MediaControlsHost::formattedStringForDuration): Deleted.
(): Deleted.
(WebCore::MediaControlsHost::showMediaControlsContextMenu): Deleted.
(WebCore::MediaControlsHost::sourceType const): Deleted.
* Source/WebCore/Modules/mediacontrols/MediaControlsHost.h:
* Source/WebCore/Modules/mediacontrols/MediaControlsHost.idl:
* Source/WebCore/Modules/modern-media-controls/media/media-controller.js:
(MediaController.prototype._updateControlsIfNeeded):

Canonical link: https://commits.webkit.org/279471@main

rkirsling commented on Source/WebCore/Modules/mediacontrols/MediaControlsHost.cpp:
```diff
@@ -315,6 +315,13 @@ bool MediaControlsHost::inWindowFullscreen() const
 
+bool MediaControlsHost::supportsRewind() const
+{
+    if (auto sourceType = this->sourceType())
```
This method doesn't exist without `ENABLE(MODERN_MEDIA_CONTROLS)`; fixing in #29293.

danae404 replied:
Thanks!
